tests/CustomerTest.php

```diff
@@ -2,337 +2,220 @@
     }
 
-    public function testCustomerUpdateCard()
+    public function testCanListInvoiceItems()
     {
-        $customer = $this->createTestCustomer();
-        $customer->save();
-
-        $sources = $customer->sources->all();
-        $this->assertSame(count($sources["data"]), 1);
-
-        $card = $sources['data'][0];
-        $card->name = "Jane Austen";
-        $card->save();
-
-        $updatedCustomer = Customer::retrieve($customer->id);
-        $updatedCards = $updatedCustomer->sources->all();
-        $this->assertSame($updatedCards["data"][0]->name, "Jane Austen");
+        $customer = Customer::retrieve(TEST_RESOURCE_ID);
+        $this->expectsRequest(
+            'get',
+            '/v1/invoiceitems',
+            array("customer" => $customer->id)
+        );
+        $resources = $customer->invoiceItems();
+        $this->assertTrue(is_array($resources->data));
+        $this->assertSame("Stripe\\InvoiceItem", get_class($resources->data[0]));
     }
 
-    public function testCustomerDeleteCard()
+    public function testCanListCharges()
     {
-        $customer = $this->createTestCustomer();
-        $createdCard = $customer->sources->create(array("card" => 'tok_visa'));
-        $customer->save();
-
-        $updatedCustomer = Customer::retrieve($customer->id);
-        $updatedCards = $updatedCustomer->sources->all();
-        $this->assertSame(count($updatedCards["data"]), 2);
-
-        $deleteStatus = $updatedCustomer->sources->retrieve($createdCard->id)->delete();
-        $this->assertTrue($deleteStatus->deleted);
-        $updatedCustomer->save();
-
-        $postDeleteCustomer = Customer::retrieve($customer->id);
-        $postDeleteCards = $postDeleteCustomer->sources->all();
-        $this->assertSame(count($postDeleteCards["data"]), 1);
+        $customer = Customer::retrieve(TEST_RESOURCE_ID);
+        $this->expectsRequest(
+            'get',
+            '/v1/charges',
+            array("customer" => $customer->id)
+        );
+        $resources = $customer->charges();
+        $this->assertTrue(is_array($resources->data));
+        $this->assertSame("Stripe\\Charge", get_class($resources->data[0]));
     }
 
-    public function testCustomerAddSource()
+    public function testCanUpdateSubscription()
     {
-        self::authorizeFromEnv();
-
-        $customer = $this->createTestCustomer();
-        $createdSource = $customer->sources->create(array("source" => 'tok_visa'));
-        $customer->save();
-
-        $updatedCustomer = Customer::retrieve($customer->id);
-        $updatedSources = $updatedCustomer->sources->all();
-        $this->assertSame(count($updatedSources["data"]), 2);
+        $customer = Customer::retrieve(TEST_RESOURCE_ID);
+        $this->stubRequest(
+            'post',
+            '/v1/customers/' . $customer->id . '/subscription',
+            array("plan" => "plan")
+        );
+        $customer->updateSubscription(array("plan" => "plan"));
```

Comment:
@ob-stripe This test (and the next one) don't test the returned value because... nothing is returned? Didn't seem to matter but I wanted to flag

Reply:
You can add a fourth parameter to `stubRequest` with the array that you want to return.

Reply:
@ob-stripe I know that, but I am testing the real function and that function did not return a resource which seems like a bug in the library itself (or potentially stripe-mock)